### ISSUE TITLE
Error during redisplay: (eval (breadcrumb--header-line)) signaled (args-out-of-range 0 1)

### ISSUE THREAD

**Issue by jcs090218** [joaotavora/breadcrumb#36]
It seems like something went wrong in `breadcrumb--ipath-plain`.

```diff
 (defun breadcrumb-ipath (index-alist pos)
   "Get breadcrumb for position POS given INDEX-ALIST."
   (if (get-text-property 0 'breadcrumb-region (caar index-alist))
       (breadcrumb--ipath-rich index-alist pos)
-     (breadcrumb--ipath-plain index-alist pos)))
+    (ignore-errors (breadcrumb--ipath-plain index-alist pos))))
```

After `ignore-errors`, the error disappear. 🤔 

### Step to reproduce

1. Open the file below in `svelte-mode` then you will see the error:

```svelte
<section>
    <h1>
        <span class="welcome">
        </span>
    </h1>

    <Counter />
</section>
```

### Additional Information

The breadcrumb is now empty due to the error. Other (or most) svelte files work fine.

**Comment by joaotavora** [comment]
Reproduced.  Maybe I could do something about this, but you should first ask `svelte-mode`s developers why the imenu structure
they they produce in your example is `(("" . 20))`, i.e. why are there menu elements with empty strings.

I.e. if I do something about this, it will not change the outcome dramatically: the error will disappear but there will be nonsensical imenu information

**Comment by leafOfTree** [comment]
Hi, I'm from svelte-mode and not quite sure about the imenu structure. svelte-mode is based on mhtml-mode.
Both have the same issue. 

It seems that the menu is using `<h1>` tag text content. The error occurs when it's empty. Otherwise, it will show its content.

```html
    <section>
      <h1>
        hello
        <span class="welcome">
        </span>
      </h1>
      <Counter />
    </section>
```

For this case, I think we can ignore the error as HTML tags may have empty text content and result in an empty menu.

**Comment by joaotavora** [comment]
> For this case, I think we can ignore the error as HTML tags may have empty text content and result in an empty menu.

Sure, I guess.  Breadcrumb operates on the same information that `M-x imenu` does.  Major mode authors setting up `imenu-create-index-function` or authors deriving from such modes such always test that `M-x imenu` does reasonable things.  

If it doesn't, neither will breadcrumb.  Imenu is a flimsy and hacked-together piece of utility, unfortunately, but it's the most widespread one.  

If the problem lies deeper in `mhtml-mode` ,then it should probably be fixed there.

**Comment by leafOfTree** [comment]
I agree the major mode (actually it's `sgml-mode` ) should provide correct and useful `imenu` info. But I'm not sure where to find the `sgml-mode` maintainer.

By the way, it seems to stop working if there is an incorrect `imenu`. The breadcrumb bar becomes empty with: Error during redisplay: (eval (breadcrumb--header-line) t) signaled (args-out-of-range 0 1)

![Screenshot 2024-05-14 at 19 23 33](https://github.com/joaotavora/breadcrumb/assets/13349592/4d643113-390b-46fd-9b51-a8ab235fdbd7)

Is it possible that, on the other side, breadcrumb can handle the error gracefully, like just ignoring the error?

**Comment by joaotavora** [comment]
> But I'm not sure where to find the sgml-mode maintainer.

Isn't it a part of Emacs?  Report it to bug-gnu-emacs@gnu.org.

> Is it possible that, on the other side, breadcrumb can handle the error gracefully, like just ignoring the error?

Yes, as I said something like that can be done. But making breadcrumb.el ignoring this particular error is somewhat tricky, so if someone has patch and can test through some edge cases (like an imenu that does have some invalid entries and some valid ones), it would be ideal.
